src/services/instance/items/index.js

```diff
@@ -244,18 +246,81 @@ module.exports = {
           );
         }
 
-        return await this.legacy.postRequest(
-          this.legacy.interpolate(this.legacy.API.publishItem, {
+        return await this.postRequest(
+          this.interpolate(this.API.publishItem, {
             MODEL_ZUID: modelZUID,
             ITEM_ZUID: itemZUID,
           }),
           {
-            usesCookieAuth: true,
             payload: {
-              version_num: version,
+              version: version,
+              publishAt: "now",
+              unpublishAt: "never",
             },
           }
         );
+
+        // return res.data;
+      }
+
+      async publishItems(items) {
+        if (!Array.isArray(items)) {
+          throw new Error(
+            "SDK:Instance:publishItems() requires `items` argument to be an array"
+          );
+        }
+        if (!items.length) {
+          throw new Error(
+            "SDK:Instance:publishItems() `items` array did not contain any items"
+          );
+        }
+
+        // GCP begins rejecting requests from a single IP
+        // after a certain threshold. 100 seems to be the magic number
+        const limit = 100;
+        const iterations = Math.floor(items.length / limit);
+
+        let results = [];
+        let run = true;
+        let index = 0;
+
+        // console.log("items: ", items.length);
+        // console.log("iterations: ", iterations);
+
+        while (run) {
+          const start = index * limit;
+          const chunk = items.slice(start, start + limit);
+
+          console.log("index: ", index);
+          // console.log("START", start);
+
+          const requests = chunk.map((item) => {
+            return this.publishItem(
+              item.meta.contentModelZUID,
+              item.meta.ZUID,
+              item.meta.version
+            );
+          });
+
+          // console.log("REQS", requests.length);
+
+          await Promise.all(requests).catch((err) => {
+            console.error(err);
+            run = false;
+          });
+
+          // capture all requests to be returned
+          results.push(...requests);
+
+          if (iterations === index) {
+            run = false;
+            break;
+          }
+
+          index++;
+        }
+
+        return results;
```

Comment:
I wonder how useful it is to have a return value for a function that takes any longer than a few minutes to reply.

Reply:
While this operation can take a long time once it is complete I would imagine the caller will want to continue handling the code flow and need to understand the result of individual requets